### PR TITLE
Fix ABSL_HAVE_ALARM check on mingw

### PR DESCRIPTION
**Pull request by jtsylve**
mingw compilers don't support alarm()

### CODE REVIEW

**Comment by EricWF** [comment]
This looks good to me.
But please be aware that  we dont have test coverage on MinGW (nor is it officially supported).

Regardless, thank you for the patch. I'll merge it once I verify it internally.

**Comment by jtsylve** [comment]
rebased against master

File: absl/base/config.h
Comment (suertreus) [inline absl/base/config.h]: Please explain in this comment why `ABSL_HAVE_ALARM` is not defined for `__MINGW32__`.  I realize you've copied the form of the preceding comments, but they're not very helpful and I'd like to do better.  The `__EMSCRIPTEN__` example below is better.

It looks to me like the right comment is:
```
// mingw32 doesn't provide alarm(2):
// https://osdn.net/projects/mingw/scm/git/mingw-org-wsl/blobs/5.2-trunk/mingwrt/include/unistd.h
// mingw-w64 provides a no-op implementation:
// https://sourceforge.net/p/mingw-w64/mingw-w64/ci/master/tree/mingw-w64-crt/misc/alarm.c
```

**Review by suertreus — CHANGES_REQUESTED** [review]

**Comment by shaindelschwartz** [comment]
@jtsylve Would you like to follow up here? There's an open review comment that should be resolved before we merge this.

**Review by suertreus — APPROVED** [review]

**Pull request merged**
